File: NoiseGenerator.java
import processing.core.PApplet;
import java.lang.Math;

public class NoiseGenerator extends Generator {

	float lastValue;
	int lastMillis;

	public NoiseGenerator(PApplet p, String name, float amp, int phase, int period) {
		super(p, name, amp, phase, period);
	}

	public float value(int millis) {
		if (millis % (period * 2) > period) {
			lastValue = (float)Math.random() * amp;
			lastMillis = millis;
		} 

		return lastValue;
	}

}

File: python/bellechasse/src/bellechasse/generators/noise_generator.py
from typing import Optional
import random
from .generator import Generator


class NoiseGenerator(Generator):
    last_value: float = 0
    last_millis: float = 0

    def __init__(
        self,
        *,
        name: Optional[str] = None,
        amp: float = 1,
        offset: float = 0,
        period: float = 500,
    ):
        super().__init__(name=name, amp=amp, offset=offset, period=period, phase=0)

    def value(self, millis: float) -> float:
        if (millis - self.last_millis) % (self.period * 2) > self.period:
            self.last_value = random.random() * self.amp + self.offset
            self.last_millis = 0

        return self.last_value
